e2e/support/test_support.go

```diff
@@ -1679,6 +1690,29 @@ func OperatorPod(ns string) func() *corev1.Pod {
 	}
 }
 
+// Find one Pod filtered by namespace ns and label app.kubernetes.io/name value appName.
+func Pod(ns string, appName string) func() *corev1.Pod {
+	return func() *corev1.Pod {
+		lst := corev1.PodList{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Pod",
+				APIVersion: v1.SchemeGroupVersion.String(),
+			},
+		}
+		if err := TestClient().List(TestContext, &lst,
+			ctrl.InNamespace(ns),
+			ctrl.MatchingLabels{
+				"app.kubernetes.io/name": appName,
+			}); err != nil {
+			failTest(err)
```

Comment:
I think it would be better to return the error and let the invoker to fail or do any other action instead of failing here.